icu4c/source/i18n/dtitvfmt.cpp

```diff
@@ -1422,7 +1422,11 @@ DateIntervalFormat::setIntervalPattern(UCalendarDateFields field,
         if ( field == UCAL_AM_PM ) {
             fInfo->getIntervalPattern(*bestSkeleton, UCAL_HOUR, pattern,status);
             if ( !pattern.isEmpty() ) {
-                setIntervalPattern(field, pattern);
+                UBool suppressDayPeriodField = fSkeleton.indexOf(CAP_J) != -1;
+                UnicodeString adjustIntervalPattern;
+                adjustFieldWidth(*skeleton, *bestSkeleton, pattern, differenceInfo,
+                                 suppressDayPeriodField, adjustIntervalPattern);
+                setIntervalPattern(field, adjustIntervalPattern);
```

Comment:
I see this is all similar to what is done later in the same function after `if ( !pattern.isEmpty() )` ...

Reply:
Yup, it's the same logic.  Since it's only a couple lines, I didn't think it was worth it to factor it out into a separate function.

Reply:
Agree, I was just noting that as a way of verifying correctness